### ISSUE TITLE
Support dynamic parameters, computed only once the test using them actually starts

### ISSUE THREAD

**Issue by vmalloc** [getslash/slash#672]

**Comment by vmalloc** [comment]
Suggested syntax:

```python

def get_parameter_value():
    return 'bla'

@slash.parametrize('param', [
    slash.param('label1', compute=get_parameter_value),
    ...
]
def test_something(param):
    # param will be 'bla'
```

**Comment by vmalloc** [comment]
Postponed to 1.7, since there is a requirement to have the parameters computed after `test_start`. This poses some complications on:

* There should be two resolve phases for parameters, the first should fill `variation.values` only with values known beforehand (non-computed). This phase should leave stubs for values that *should* be computed, so that plugins registered on `test_start` can inspect `variation.values` and detect future computations. The second phase should fill these gaps up
* `log.show_raw_param_values` should display these gaps in a meaningful way (e.g. `<computed:func_1>` or something). If it is enabled, and at least one dynamic parameter is encountered, another info log should be emitted with the final values of the computed params
* An additional hook should be added for after the parameters are resolved, which will be used by Backslash to fill in the computed values

**Comment by maormarcus** [comment]
- Exclusions on computed parameters should cause runtime error when encountered

**Comment by vmalloc** [comment]
Rejected in favor of using fixtures
